### PR TITLE
Fix version format regexp

### PR DESCRIPTION
**Pull request by asamuzaK**
The current TOOLKIT_REGEX regular expression has a bug.
* For example, 1.00 is invalid but 1.00a1 is marked as valid.
* It treats other string than a|alpha|b|beta|pre|rc plus optional integer as valid.

This PR fix them.

NOTE: '0.1.12dev-cb31c51', '4.1.1dev-abcdef1' will be invalid in this PR.
Please don't merge unless decision is made that those strings should be invalid.
See [#1439 (comment)](https://github.com/mozilla/addons-linter/issues/1439#issuecomment-324538056 "Error message for versions like 1.4.0-pre.0") and [#1439 (comment)](https://github.com/mozilla/addons-linter/issues/1439#issuecomment-324712461 "Error message for versions like 1.4.0-pre.0")

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
This looks great so far, I'll dig a bit deeper into the regexes later today and check them against a few more live add-ons

> NOTE: '0.1.12dev-cb31c51', '4.1.1dev-abcdef1' will be invalid in this PR.
Please don't merge unless decision is made that those strings should be invalid.

That decision was made in the comments you linked. Firefox won't be able to parse them properly - or rather might allow them but doesn't properly update - so we should forbid them.

**Comment by asamuzaK** [comment]
Added tests for isToolkitVersionString()

File: src/schema/formats.js [hunk 1]
```diff
@@ -1,48 +1,32 @@
...
-    return false;
-  }
-
-  for (let i = 0; i < parts.length; i++) {
-    let part = parts[i];
-    // Leading or multiple zeros not allowed.
-    if (part.startsWith('0') && part.length > 1) {
-      return false;
-    }
-    // Disallow things like 123e5 which parseInt will convert.
```
Comment (EnTeQuAk) [inline src/schema/formats.js]: This warrants a test, can you add that?

Comment (asamuzaK) [reply src/schema/formats.js]: Done.
And also added more test patterns.

Comment (EnTeQuAk) [reply src/schema/formats.js]: Ah ` '123e5'` in the invalid patterns, somehow I missed that.

Nice, thanks!

**Comment by EnTeQuAk** [comment]
I just merged this, thank you so much for taking care of this!

**Pull request merged**